README.md

```diff
@@ -6,3 +6,6 @@
 [![Mac Build](https://github.com/mitchute/portable_component_model/actions/workflows/mac_build.yml/badge.svg)](https://github.com/mitchute/portable_component_model/actions/workflows/mac_build.yml)
 
 The portable component model outlines the methods needed to define a standalone component model, and then export that model to be used as either a FMU via the FMI interface or included in EnergyPlus.
+
+## Structure
+![structure](images/structure.png)
```

Comment:
I'd add some details to the diagram like:
- What does each circle build into, like EXE, DLL, etc.?
- Can the model be built as a static library to be consumed at compile time, or just a dynamic library?
- On the arrows, are those runtime calls, or baked into function calls at runtime?

It may actually be good to make three separate boxes, one for each application:

```
                                                                 
******************   ******************    ********************* 
*        *       *   *        *       *    *          *        * 
*        *       *   *        *       *    *          *        * 
*    ********    *   *    ********    *    *     *********     * 
*    *   *  *    *   *    *   *  *    *    *     *    *  *     * 
*    *STATIC*    *   *    *STATIC*    *    *     * DLL   *     * 
*    ********    *   *    ********    *    *     *********     * 
*                *   *                *    *                   * 
*                *   *                *    *                   * 
******************   ******************    ********************* 
                                                                 
  ENERGYPLUS             FMU                   STANDALONE        
                                                                 
     EXE                 DLL                      EXE            
                                                                 
```